.github/workflows/ci.yml

```
@@ -154,3 +167,181 @@ jobs:
+        if: matrix.tests == 'python-unit'
+
+      - name: Install functional test requirements
+        run: |
+          python -m pip install numpy
+          python -m pip install --upgrade -r ./build/functional-test-requirements.txt
+        if: matrix.tests == 'functional'
+
+      - name: Prepare environment for venv tests
+        env:
+          TEST_FILES_SUFFIX: testvirtualenvs
+          PYTHON_VIRTUAL_ENVS_LOCATION: './src/tmp/envPaths.json'
+        run: |
+          # venv/bin or venv\\Scripts (windows)?
+          $environmentExecutableFolder = 'bin'
+          if ('${{runner.os}}' -match 'windows-latest') {
+            $environmentExecutableFolder = 'Scripts'
+          }
+
+          # pipenv
+          python -m pip install pipenv
+          python -m pipenv run python build/ci/addEnvPath.py ${{env.PYTHON_VIRTUAL_ENVS_LOCATION}} pipenvPath
+
+          # venv
+          if ('${{matrix.python}}' -like '3.?') {
+            python -m venv .venv
+            & ".venv/$environmentExecutableFolder/python" ./build/ci/addEnvPath.py ${{env.PYTHON_VIRTUAL_ENVS_LOCATION}} venvPath
+          }
+
+          # virtualenv
+          python -m pip install virtualenv
+          python -m virtualenv .virtualenv
+          & ".virtualenv/$environmentExecutableFolder/python" ./build/ci/addEnvPath.py ${{env.PYTHON_VIRTUAL_ENVS_LOCATION}} virtualEnvPath
+
+          # conda
+
+          # 1. For `terminalActivation.testvirtualenvs.test.ts`
+
+          $condaExecPath = Join-Path -Path $Env:CONDA -ChildPath $environmentExecutableFolder | Join-Path -ChildPath conda
+          if ('${{runner.os}}' -match 'windows-latest') {
+              $condaPythonPath = Join-Path -Path $Env:CONDA -ChildPath python
+          } else{
+              $condaPythonPath = Join-Path -Path $Env:CONDA -ChildPath $environmentExecutableFolder | Join-Path -ChildPath python
+              & $condaPythonPath ./build/ci/addEnvPath.py ${{env.PYTHON_VIRTUAL_ENVS_LOCATION}} condaExecPath $condaExecPath
+          }
+          & $condaPythonPath ./build/ci/addEnvPath.py ${{env.PYTHON_VIRTUAL_ENVS_LOCATION}} condaPath
+
+          # 2. For `interpreterLocatorService.testvirtualenvs.ts`
+
+          & $condaExecPath create -n "test_env1" -y python
+          & $condaExecPath create -p "./test_env2" -y python
+          & $condaExecPath create -p "$Env:HOME/test_env3" -y python
+        shell: pwsh
+        if: matrix.tests == 'venv'
+
+      # Run TypeScript unit tests only for Python 3.X.
+      - name: Run TypeScript unit tests
+        run: npm run test:unittests:cover
+        if: matrix.tests == 'ts-unit' && startsWith(matrix.python, 3.)
+
+      # Run the Python IPython tests in our codebase.
+      # Produce JUnit-style log files that will be uploaded after all tests are complete.
+      - name: Run Python and IPython unit tests
+        run: |
+          python pythonFiles/tests/run_all.py --color=no --junit-xml=${{env.TEST_RESULTS_DIRECTORY}}/python-tests-junit.xml
```

Is `--color=no` a hold-over from Azure Pipelines, or did you find that out fresh for GH Actions?

```suggestion
          python pythonFiles/tests/run_all.py --junit-xml=${{env.TEST_RESULTS_DIRECTORY}}/python-tests-junit.xml
```

---

Copy- 🍝 from AzDO